src/notifiers/pool.js

```diff
@@ -0,0 +1,42 @@
+import _s from 'underscore.string';
+import glob from 'glob';
+import fs from 'fs';
+import path from 'path';
+
+import NotifierLoader from './loader';
+
+let _instance = null;
+
+export default class NotifierPool {
+  constructor() {
+    if (!_instance) {
```

Comment:
bit tricky but you can do `this.constructor._instance` which works like static variable